stats/cloud/collector.go

```diff
@@ -51,6 +51,9 @@ type Collector struct {
 
 	sampleBuffer []*Sample
 	sampleMu     sync.Mutex
+
+	runStatus   int
```

Comment:
I think that if you make this `chan int` it would have two advantages while having the external interface (`SetRunStatus(status int)`) stay the same:
1. You won't need an extra mutex, just set it with `c.runStatus <- status` and read it with `runStatus := <- c.runStatus`
2. Right now it seems that there's a small chance for a race condition in the sense that `testFinished()` may be finished before `SetRunStatus()` is called. If you use a channel, `testFinished()` will wait if necessary and get the correct status for sure.